librz/include/vmlinux.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
```

Comment:
1. SPDX
2. Doxygen comments for added structures and the purpose of this file

Reply:
please don't use `#pragma once` but use proper `#ifndef XXX` `#define XXX` also add the c++ guards.

Reply:
added doxygen & spdx. usnig `#ifndef` now.

Reply:
@wargio could you please elaborate more on c++ guards? what should be added?